src/main/java/com/github/maxopoly/environmentaleffects/datarepresentations/MobConfig.java

```diff
@@ -142,8 +145,10 @@ public MobConfig(String identifier, EntityType type, String name,
 
 	public Location findSpawningLocation(Location loc) {
```

Comment:
In general the plugin will loop over all the players online and try to spawn mobs next to them. If the random number chosen every loop is lower than the spawn chance of a specific mob the plugin will call this method and attempt to find a location to spawn the mob in. This method will choose a random X and Z coordinates near the player and then scan from bedrock to sky limit in attempt to find a spot the mob can fit into. It will save all the possible y levels the mob can spawn in and then choose a random one of them. I guess I can start adding comments so you have an understanding of what each part of the code does but this is a bit of a waste of time considering there is already a complete new algorithm max coded for Genesis(https://github.com/Maxopoly/Genesis/blob/master/src/main/java/com/github/maxopoly/Genesis/spawning/SpawnFinder.java) that is going to replace what we have now in EE.